src/js/helpers/checkboxTreeHelper.js

```diff
@@ -57,7 +57,7 @@ export const removePlaceholderString = (str) => {
 };
 
 export const getAllDescendants = (node) => {
-    if (!node.children) return [node.value];
+    if (!node.children || node?.children?.length === 0) return [node.value];
```

Comment:
Adding extra case here to ensure that we're also returning `[node.value]` if the children prop is defined, but actually just an empty array.